### DISCUSSION TITLE
Why do I need the windows.iso installation file and drive if the VM is already set up?

### DISCUSSION THREAD

**Discussion by Satlinker** [virt-manager/virt-manager#646] [Q&A]
Hi,
I try to install a win10 VM in Linux. It works, but virt-manager doesn't start without the usb-stick with windows.iso file on it. Why do I need the windows.iso if windows is already installed?

**Comment by phrdina** [comment]
Hi, how did you configure the iso image for the VM? Can you please provide the VM XML and the error message you are getting?

**Comment by Satlinker** [comment]
I had to add both drives to pools:
The windows.iso is located on an usb-stick on: /run/media/saxli/DEBIAN
The virtual machine is located on a separate ext4 partition: /run/media/saxli/264627C546279519/Virt_Win10

Error if usb stick is unplugged:
Fehler beim Starten der Domain: Cannot access storage file '/run/media/saxli/DEBIAN/Windows.iso': No such file or directory
Traceback (most recent call last):
  File "/usr/share/virt-manager/virtManager/asyncjob.py", line 72, in cb_wrapper
    callback(asyncjob, *args, **kwargs)
  File "/usr/share/virt-manager/virtManager/asyncjob.py", line 108, in tmpcb
    callback(*args, **kwargs)
  File "/usr/share/virt-manager/virtManager/object/libvirtobject.py", line 57, in newfn
    ret = fn(self, *args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/share/virt-manager/virtManager/object/domain.py", line 1402, in startup
    self._backend.create()
  File "/usr/lib/python3.11/site-packages/libvirt.py", line 1379, in create
    raise libvirtError('virDomainCreate() failed')
libvirt.libvirtError: Cannot access storage file '/run/media/saxli/DEBIAN/Windows.iso': No such file or directory

**Comment by phrdina** [comment]
When virt-manager installs a new VM in creates two VM definitions, one that is only used to start the VM for first time in order to install it and second that is saved persistently. For most OSes once the installation is done it usually reboots the system. The first VM definition is set to completely power off when reboot happens and the VM is started again using the persistent definition.

Now with linux OSes they no longer need the installation ISO after the first reboot so the persistent definition doesn't contain the ISO image, only empty CD-ROM device. For Windows the situation used to be different and it required the ISO image even after the first reboot so we could not use the same trick to remove the ISO image. I did a simple test and it seems that at least windows 11 no longer requires the ISO image after the first reboot so we could possible improve the default behavior.

To fix this for your VM go to details of the VM and remove the CD-ROM or the `Source path` if you want to keep the CDROM device available.

**Reply by Satlinker** [reply]
Thank you Pavel,
>To fix this for your VM go to details of the VM and remove the CD-ROM or the Source path if you want to keep the CDROM device available.

It only works as long as I don't close virt-manager. When I shut all down and open it again, virt-manager wants the USB stick again, although it is no longer in the storage pool.
